src/codeflare_sdk/cluster/cluster.py

```diff
@@ -268,19 +268,27 @@ def wait_ready(self, timeout: Optional[int] = None):
         dashboard_ready = False
         status = None
         time = 0
-        while not ready or not dashboard_ready:
+        while not ready:
             status, ready = self.status(print_to_console=False)
-            dashboard_ready = self.is_dashboard_ready()
             if status == CodeFlareClusterStatus.UNKNOWN:
                 print(
                     "WARNING: Current cluster status is unknown, have you run cluster.up yet?"
                 )
-            if not ready or not dashboard_ready:
+            if not ready:
+                if timeout and time >= timeout:
+                    raise TimeoutError(f"wait() timed out after waiting {timeout}s for cluster to be ready")
+                sleep(5)
+                time += 5
+        print("Requested cluster is up and running!")
+
+        while not dashboard_ready:
```

Comment:
While we are separating the two, could we also add a new argument for wait_ready like `dashboard_check: bool=True` (default true) and have this line be like:
```
while dashboard_check and not dashboard_ready:
```
Just for kubernetes folks who may not have/want a dashboard route currently

Reply:
Great idea, that's added there now. One thought on that though, the `while dashboard_check and not dashboard_ready` is checking if the dashboard is ready if the `dashboard_check` is true. But, whether it's true or false, the route will be generated and can be seen when running i.e., `cluster.details()`. Is this something to assess in this PR or in a separate one?

Reply:
Yeah this is fine as-is, once Mark's PR on the KubeRay side is complete we will revisit the route itself from the SDK